[IMP] auditlog - Autovacuum logs

sebalix opened the pull request:
Add a `ir.cron` to delete logs periodically (fixing #187).

The index on the `log_id` field on the `auditlog_log_line` table is required to delete thousands of logs almost instantaneously.

I will bump the version number after the PR #357 is merged.

sebalix committed:
[IMP] auditlog - Auto-vacuum logs, HTTP requests and HTTP user sessions

hbrunn commented:
:+1: , but adding deletion to tests would be nice

sebalix commented:
@hbrunn I will do that

sebalix commented:
Thinking about it, I will rewrite this to purge user sessions and HTTP requests records too.

sebalix commented:
Ready for review.

elicoidal commented on auditlog/README.rst:
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
 operations to log for a given data model.
```
Hi @sebalix Thanks for the contribution.
I miss the top of the README file: badge + line for document title.

sebalix replied:
Fixed.

elicoidal commented on auditlog/__openerp__.py:
```diff
@@ -21,7 +21,7 @@
 {
     'name': "Audit Log",
-    'version': "8.0.1.2.0",
+    'version': "8.0.1.3.0",
```
According to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers I would tend to have 8.0.2.0.0  (new views and new data in models).
What do you think?

sebalix replied:
I checked this document, and as it is a non-breaking new feature which does not require a data migration, I opted for `1.3.0`. This feature is less intrusive than #357 where I have done the same incrementation. But it does not matter for me if it has to be `2.0.0`, what other peoples think?

elicoidal replied:
Finally not so sure: 8.0.1.3.0 looks good...

elicoidal commented on auditlog/data/ir_cron.xml:
```diff
@@ -0,0 +1,18 @@
+    <data noupdate="1">
+
+        <record id="ir_cron_auditlog_autovacuum" model="ir.cron">
+            <field name='name'>AutoVacuum audit logs</field>
```
s/AutoVacuum/Auto-vacuum

sebalix replied:
I just copied https://github.com/OCA/connector/blob/9.0/connector/queue/queue_data.xml#L31 to be identical with others and have a nice and coherent overview of the scheduled actions in the tree view:
- AutoVacuum audit logs
- AutoVacuum osv_memory objects
- AutoVacuum Queue Jobs

But your suggestion is right.

elicoidal replied:
Your call: this is not so important if it is consistent with existing "misspelling" ;)

sebalix replied:
So be it, fixed ;)

elicoidal commented on auditlog/models/log.py:
```diff
@@ -18,7 +18,6 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-
```
Can you take the opportunity to simplify all headers from py files?

sebalix replied:
I agree, but I would like to do these kind of changes in another PR (simplify headers, add screenshots, improve/clean up the README...).

elicoidal replied:
OK for me

elicoidal commented on auditlog/README.rst:
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
+`Configuration / Technical / Automation / Scheduled Actions` menu and edit the
+`AutoVacuum audit logs` task parameters.
+
 During installation, it will migrate any existing data from the `audittrail`
```
During installation, a one-time script will migrate any existing data from the `audittrail`
 module (rules and logs).

sebalix replied:
Fixed.

elicoidal commented on auditlog/README.rst:
```diff
@@ -11,6 +11,11 @@ Go to `Reporting / Audit / Rules` to subscribe rules. A rule defines which
+
 During installation, it will migrate any existing data from the `audittrail`
 module (rules and logs).
 
```
Remove the part below on the forum, not necessary anymore.

elicoidal replied:
in the url below: https (x2)

sebalix replied:
Fixed.

elicoidal commented on auditlog/__openerp__.py:
```diff
@@ -21,7 +21,7 @@
 
```
simplify header

elicoidal commented on auditlog/README.rst:
```diff
@@ -1,5 +1,8 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Audit Log - Track user operations
```
you need to add one top line being document title

sebalix replied:
Oh ok, fixed.

elicoidal commented:
LGTM

sebalix commented:
The unit tests seems to fail sometimes, it is surely related to the date comparison in the search pattern... Do not merge yet, WIP

sebalix commented:
Fixed, I added a `time.sleep(1)` in the unit test before the call to `autovacuum(days=0)` (https://github.com/OCA/server-tools/pull/360/files#diff-ca82e7cf6716c1ba4ff7d52084a0e14fR42).

StefanRijnhart commented on auditlog/data/ir_cron.xml:
```diff
@@ -0,0 +1,18 @@
+            <field name='interval_number'>1</field>
+            <field name='interval_type'>days</field>
+            <field name="numbercall">-1</field>
+            <field name="active">True</field>
```
Set this to False by default, so that it does not suprise any current users that their logs start to be wiped after half a year?

sebalix replied:
Done, and README updated.

sebalix commented:
2 thumbs up + more than 5 days, I think it can be merged? Someone available to do it? ;)

pedrobaeza commented:
OK, merging.

sebalix commented:
@pedrobaeza thank you!
